core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/WorkflowVersionResource.scala

```diff
@@ -0,0 +1,145 @@
+package edu.uci.ics.texera.web.resource.dashboard
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.flipkart.zjsonpatch.{JsonPatch}
+import edu.uci.ics.texera.web.SqlServer
+import edu.uci.ics.texera.web.model.jooq.generated.Tables.{WORKFLOW, WORKFLOW_VERSION}
+import edu.uci.ics.texera.web.model.jooq.generated.tables.daos.{WorkflowDao}
+import edu.uci.ics.texera.web.model.jooq.generated.tables.pojos.{Workflow}
+import edu.uci.ics.texera.web.resource.auth.UserResource
+import edu.uci.ics.texera.web.resource.dashboard.WorkflowResource.context
+import io.dropwizard.jersey.sessions.Session
+import org.glassfish.jersey.media.multipart.FormDataParam
+import org.jooq.types.UInteger
+
+import java.sql.Timestamp
+import javax.servlet.http.HttpSession
+import javax.ws.rs._
+import javax.ws.rs.core.{MediaType, Response}
+import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
+
+/**
+  * This file handles various request related to workflows versions.
+  * The details of the mysql tables can be found in /core/scripts/sql/texera_ddl.sql
+  */
+
+case class VersionEntry(
+    vId: UInteger,
+    creationTime: Timestamp,
+    content: String
+)
+
+object WorkflowVersionResource {
+  val context = SqlServer.createDSLContext()
+}
+
+@Path("/version")
+@Produces(Array(MediaType.APPLICATION_JSON))
+class WorkflowVersionResource {
+
+  final private val workflowDao = new WorkflowDao(context.configuration)
+
+  /**
+    * This method returns the versions of a workflow given by its ID
+    *
+    * @param session HttpSession
+    * @return versions[]
+    */
+  @GET
+  @Path("/{wid}")
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  def retrieveVersionsOfWorkflow(
+      @PathParam("wid") wid: UInteger,
+      @Session session: HttpSession
+  ): List[VersionEntry] = {
+    UserResource.getUser(session) match {
+      case Some(user) =>
+        val versions = context
+          .select(WORKFLOW_VERSION.VID, WORKFLOW_VERSION.CREATION_TIME, WORKFLOW_VERSION.CONTENT)
+          .from(WORKFLOW_VERSION)
+          .leftJoin(WORKFLOW)
+          .on(WORKFLOW_VERSION.WID.eq(WORKFLOW.WID))
+          .where(WORKFLOW_VERSION.WID.eq(wid))
+          .fetch()
+        versions
+          .map(workflowRecord =>
+            VersionEntry(
+              workflowRecord.into(WORKFLOW_VERSION).getVid,
+              workflowRecord.into(WORKFLOW_VERSION).getCreationTime,
+              workflowRecord.into(WORKFLOW_VERSION).getContent
+            )
+          )
```

Comment:
maybe use `fetchInto[VersionEntry]` to construct the custom entity `VersionEntry`?